libraries/Microsoft.Bot.Builder/BotStateSet.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Builder
+{
+    /// <summary>
+    ///  Middleware that will call `read()` and `write()` in parallel on multiple `BotState` 
+    ///  instances.
+    /// </summary>
+    /// <remarks>
+    ///     This example shows boilerplate code for reading and writing conversation and user state within
+    ///      a bot:
+    ///     
+    ///      ```JavaScript
+    ///      const { BotStateSet, ConversationState, UserState, MemoryStorage } = require('botbuilder');
+    ///     
+    ///      const storage = new MemoryStorage();
+    ///      const conversationState = new ConversationState(storage);
+    ///      const userState = new UserState(storage);
+    ///      
+    ///      adapter.use(new BotStateSet(conversationState, userState));
+    ///     
+    ///      server.post('/api/messages', (req, res) => {
+    ///      adapter.processActivity(req, res, async (context) => {
+    ///            // ... route activity ...
+    ///     
+    ///     
+    ///  ```
+    /// </remarks>
+    public class BotStateSet : IMiddleware
+    {
```

Comment:
I'm looking for a corresponding "BotStateSetTests" file to see how this works but don't see it. 

Please create a corresponding test file.